File: searching/linear_search.rb
def linear_search(arr, key)
  arr.each_with_index do |el, index|
    if el == key
      return index
    end
  end
  return -1
end

arr = Array.new
puts "Enter the size of Array"
n = gets.chomp().to_i

puts ("Enter the " + n.to_s + " elements in the array.")
for i in 0..(n)-1
  x = gets.chomp()
  arr.push(x)
end

puts "Enter the element to be searched : "
key = gets.chomp()

result = linear_search(arr, key)

result != -1 ? (puts "Element found at location " + result.to_s) : (puts "Element not found")

File: searching/python/Linear_search.py
def linearSearch(arr, n, x): 
	for i in range (0, n): 
		if (arr[i] == x): 
			return i
	return -1


arr = [2, 3, 4, 10, 40]
x = 10 
n = len(arr)
result = linearSearch(arr, n, x) 
if(result == -1): 
	print("Element is not present in array") 
else: 
	print("Element is present at index", result)
